Fixes and improvements for moodle 3.1

rogerbaba opened the pull request:
- fixed wrong instance_id in table groupselect_groups_teachers
- backup and restore works now
- added new logging events for adding non-editing teacher and creating a download link
- some small fixes and removed deprecated code
- tested with moodle version 2.9 and 3.1

pauln committed:
mod/groupselect: Omit groups/groupings which were created by the Allocate+ group sync

pauln committed:
Make 1.9 backup converter set intro format to HTML rather than Moodle

pauln committed:
Add new SVG and 24px PNG icon

pauln committed:
Fix pluginfile URLs in module intro

Was rewriting pluginfile URLs with "mod_page" instead of "mod_groupselect"

JimThorpeNZ committed:
replace calls to deprecated get_context_instance()

JimThorpeNZ committed:
Migrate to new logging API.

JimThorpeNZ committed:
Implemented new version for 'view all' logging

JimThorpeNZ committed:
Fixed issue with loggin 'viewall'

pauln committed:
Update version.php for Moodle 3.0 compat

Moodle 3.0 requires the use of $plugin rather than $module.

pauln committed:
Remove module-specific requiremodintro

Newer versions of Moodle have shifted to a single site-wide
"requiremodintro" setting, instead of module-specific ones.

pauln committed:
Use standard_intro_elements instead of add_intro_module

Use moodleform_mod::standard_intro_elements instead of
moodleform_mod::add_intro_module, as the latter has been deprecated.

Commit:
Update settings.php

get_string('configrequiremodintro', 'admin') entfernt da nicht existent

Commit:
Update settings.php

added : get_string('requiremodintro_desc', 'admin')

Commit:
add_intro_editor() replaced with standard_intro_elements()
because 'requiremodintro' is not supported anymore in the activity
settings

MDL-49101: 
Add an admin setting config->requiremodintro to force activity
descriptions always or never

Commit:
deleted general settings
because 'requiremodintro' is not supported anymore in the activity
settings

MDL-49101: 
Add an admin setting config->requiremodintro to force activity
descriptions always or never

Commit:
2016.05.09: Removed general setting 'requiremodintro', not supported
anymore by moodle

Commit:
new version of plugin an requred moodel version

Commit:
Error solved get group members if a group is being left from a user
removed callback function from confirm_action.

Commit:
some issues solved of deprecated code and declare it for stable

Commit:
Backup of teachers and passwords implemented

Commit:
restore of teachers and passwords implemented

Commit:
backup enabled

Commit:
correct instance_id of the table groupselect_groups_teachers

Commit:
Correction of assigned teacher selection

Commit:
Uppercase placeholder

Commit:
export from wrong table name
incorrect function call
remove intance id

Commit:
Uppercase placeholder

Commit:
extended with set_mappings

Commit:
new event class definition

Commit:
added events for logging

Commit:
update incorrect instance_id in table groupselect_groups_teachers

Commit:
new events

Commit:
bugfxes

Commit:
new stable version

Commit:
new stable version

Commit:
new stable version

Commit:
Update README.md

notes updated

Commit:
Update README.md

notes updated

Commit:
new option to define group name for students
new option to avoid that students can define passwords
fixed validation messages for creat group
fixed install.xml

Commit:
alpha sorted

Commit:
Merge branch 'master' of https://github.com/BLC-HTWChur/moodle-mod_groupselect.git

Commit:
new option to disable notifications if open until date is reached

Commit:
new option to disable notifications if open until date is reached

Commit:
new description for new release 3.09

Commit:
new description for new release 3.09

Commit:
small fix

Commit:
small fix

Commit:
change: table-column width not static any more

1. delete static css values in php generated html
2. add css class and add to htw theme
3. edit button description in table view: repetition of groupname not
necessary

Commit:
change: table-column width not static any more

1. delete static css values in php generated html
2. add css class and add to htw theme
3. edit button description in table view: repetition of groupname not
necessary

Commit:
feature show description activated
version increased

Commit:
feature show description activated
version increased

Commit:
Additional 

Update description for 3.10

Commit:
Additional

Update description for 3.10

Commit:
Fix: narrow table widths and line breaks in table rows

Commit:
Fix: narrow table widths and line breaks in table rows

Commit:
beta info removed from module help intro

Commit:
beta info removed from module help intro

marinaglancy commented:
Hi Rog,
this plugin is no longer maintained by Pirkka at https://moodle.org/plugins/mod_groupselect . I have not noticed your pull request (don't know why) and pretty much repeated half of your changes: https://github.com/birrel/moodle-mod_groupselect/compare/master...marinaglancy:wip-fixes

But looking at all the work that you've done I think that you would be a better person to take over maintenance of this plugin. Would you be interested to do it? If so, can you please contact me on https://moodle.org/user/profile.php?id=1334243 

Thanks in advance
Marina

marinaglancy committed:
remove trailing spaces

marinaglancy committed:
specific CSS styles

marinaglancy committed:
export only current grouping if specified

marinaglancy committed:
corrections to backup

marinaglancy committed:
Release 3.11 from 2016-09-08

marinaglancy committed:
new repo link

marinaglancy commented:
Thank you very much!
This pull request was included in the latest version released in https://moodle.org/plugins/mod_groupselect
and is available in the git repository:
https://github.com/marinaglancy/moodle-mod_groupselect

I have added a commit on top fixing some little mistakes in backup code (https://github.com/marinaglancy/moodle-mod_groupselect/commit/b68be016498b8747064a7241be97f0ba03df5bc7) and also corrections to other issues reported in this tracker.

marinaglancy committed:
fix files in description

marinaglancy committed:
Correctly process calendar events

Commit:
Merge remote-tracking branch 'refs/remotes/marina/master'

jerryshan committed:
updated with latest changes

jerryshan committed:
mod/groupselect: added default setting options.

jerryshan committed:
student group options are now changed to be enabled/disabled depends on the Student can create groups option

jerryshan committed:
fixed typo

lucaboesch committed:
Added SVG and PNG image following https://docs.moodle.org/dev/Moodle_icons guidelines.

Commit:
Merge pull request #5 from lucaboesch/master

Added new SVG and PNG activity icon

Commit:
corrections and added version info

Corrected and added version info for Project page

Commit:
Verions update

Commit:
configurable settings

Commit:
Whitespacing

Commit:
update note

Commit:
Header of action column

Commit:
Removed wrong merged line and cleaned withespace, Version update

lucaboesch committed:
Fix for #10.

Removed unused calls to https://github.com/BLC-HTWChur/moodle-mod_groupselect/blob/master/view.php which threw PHP Notices.

Signed-off-by: Luca Bösch <luca.boesch@bfh.ch>

Commit:
Merge pull request #11 from lucaboesch/master

Fix for #10.

Commit:
Added Travis yml

Commit:
Merge branch 'master' of https://github.com/BLC-HTWChur/moodle-mod_groupselect

Commit:
Code cleanup

phish108 committed:
initial role selection version

phish108 committed:
remove role configuration from instance configuration

phish108 committed:
clear module administrator

phish108 committed:
handle database requests for default roles; fixes #14

phish108 committed:
fixes #13

phish108 committed:
database structure for multiple groups

phish108 committed:
Frontend multi Group Selection; fixes #8

phish108 committed:
Merge branch 'issue8'

phish108 committed:
fixes #18

phish108 committed:
Merge branch 'issue_17'

phish108 committed:
better error messages

phish108 committed:
change report

phish108 committed:
roll back 1 as lowest max group size

Commit:
travis file correct name

Commit:
remove old travis

Commit:
fix codeguideline issues; version update;

Commit:
travis moodle check vor version 3.3

Commit:
updated travis environment

Commit:
fix issues #21 #22 #23 #24 #25 #26

Commit:
fix Remove must exist "teacher" role #31

Commit:
fix #37 #36 in validation

Commit:
fix #41; fix #39

Commit:
fix #35

Commit:
fix #40 Error "node.getAttribute is not a function" in the sortable.js

Commit:
fix #40 Error "node.getAttribute is not a function" in the sortable.js

Commit:
Merge branch 'master' of https://github.com/BLC-HTWChur/moodle-mod_groupselect

Commit:
fix #33 completion date events; fix #35 course overview timeline

Commit:
Add button to manage groups in the settings navbar #42

Commit:
Remove unnessesary code for assign supervisonrole

Commit:
#42 sets the context and course variable correctly for the group button

Commit:
#43 Add a function to unassign supervisors from groups

Commit:
cleanup

Commit:
remove the weird yellow background color of the download link

Commit:
new translation strings for unassign supervisors #43

Commit:
Event handler for #1 remove supervisors, if they got unenrolled

Commit:
fix #32 Allow to create multipe groups and fix language strings

Commit:
fix #45 New settings to join or leave groups

Commit:
new Eventhanlder for deleting groups

Commit:
fix #44 implementing reset of the group select

Commit:
Added changes for version 3.14

Commit:
added comment block and internal moodle

Commit:
Extend travis ci check ignore bad js files and support moodle 3.4

Commit:
fix #48 update capabilities for teachers, editingteachers and managers

Commit:
increment version for fix # 48

Commit:
fix #47 Rearrange settings

Commit:
Update changes and version

Commit:
fix comments, documentation and codestyle errors, inc version build nr.

Commit:
update default capabilities for fix #48

Commit:
fix #50 missing filelds in backup: supervisionrole, maxgroupmembership

Commit:
adding new fields for #45 New settings to join or leave groups

Commit:
add comment

Commit:
Adding changed and new version number

Commit:
Adding changes and new version number

Commit:
Merge branch 'master' of https://github.com/BLC-HTWChur/moodle-mod_groupselect

Commit:
fix get role ids

Commit:
fix role id ->

Commit:
fix #52 Implementing the interface of the privacy api for GDPR

Commit:
fix build number and required moodle

Commit:
Adding Branch 35 for travis

Commit:
Remove php 5.6 from Travis and add php 7.2

Commit:
comment corection

Commit:
remove not deprecated string file as it is no longer reguired

rogerbaba committed:
fix #58 show group image

rogerbaba committed:
fix #59 Enable auto completion rule viewed

lucaboesch committed:
Provide a 'More help' link to MoodleDocs in activity chooser.

Commit:
issue#97: add limits per group

Add a new table
* Store groupselectid and groupid
* Store limit per group

Add view tabs
* Tabs.php for the tabs
* Update land strings for tabs

Add Form and limit page

Update Form
* Use groupid as key and limit as value

Update form and handler
* Fix according to Moodle coding standard

Add to limits table
* Add SQL to insert limit to table
* Update records if exists

Update view logic
* Add conditions to check for adding infinite members

Add override group limit permission

Add backup and restore functionality.

Commit:
Fixed upgrade.php to match install.xml

tlock committed:
Optionally hide suspended students.

 - Hide suspended users when the setting is enable to allow active students to enroll in the groups.

golenkovm committed:
Update core_calendar_provide_event_action() method to meet MDL-58768, MDL-64783 reqs #70

rogerbaba committed:
fix #65 adjust the precision of minmembers

rogerbaba committed:
typo in function call

rogerbaba committed:
Additional implementation for MDL-58768

rogerbaba committed:
Add option hide suspended users

rogerbaba committed:
fix #19 Option to hide group members

rogerbaba committed:
Release 3.16

rogerbaba committed:
add fix

rogerbaba committed:
travis update

rogerbaba committed:
add postgresql

rogerbaba committed:
code cleanup and documentation

rogerbaba committed:
show group members always for group managers

rogerbaba committed:
Adjust php version

rogerbaba committed:
Merge branch 'master' of https://github.com/rogerbaba/moodle-mod_groupselect

rogerbaba committed:
adjust php exclude

rogerbaba committed:
added check and migrate to blackboard

rogerbaba committed:
change matrix

rogerbaba committed:
test

rogerbaba committed:
remove shifter and js lint; add mustache

rogerbaba committed:
adjust width of group description

rogerbaba committed:
fix #73 Add a direct linkt to manage groups

rogerbaba committed:
Merge branch 'master' of https://github.com/rogerbaba/moodle-mod_groupselect

rogerbaba committed:
Added Feature and fix for 3.16

lucaboesch committed:
Take hidegroupmembers from activity setting.

lucaboesch committed:
Text in modulename_help needs Markdown.

rogerbaba committed:
Fix hide groupembers only when set

rogerbaba committed:
New version for fix

rogerbaba committed:
add a line break

lucaboesch committed:
Some modest style amendments.

lucaboesch committed:
Added PURPOSE and new icon for Moodle 4.

lucaboesch committed:
Let README.md feature a status pill.

rogerbaba committed:
typo

lucaboesch committed:
Activity completion integration, resolves #56.

rogerbaba committed:
remove header from older moodle versions

rogerbaba committed:
Merge branch 'master' of https://github.com/rogerbaba/moodle-mod_groupselect

rogerbaba committed:
remove older moodle version code

lucaboesch committed:
Increase version in moodle-plugin-ci.yml.

lucaboesch committed:
Adding activity dates information fixes #89. (#92)

Co-authored-by: Rog <17590070+rogerbaba@users.noreply.github.com>

rogerbaba committed:
use activity date translations from course

rogerbaba committed:
fix #82 Audit $OUTPUT notifications

rogerbaba committed:
Remove douplicate info for moodle >=311

rogerbaba committed:
Removing the XSS Risk as input is cleaned #80

rogerbaba committed:
fix #78 setting group description always enabled

rogerbaba committed:
new Release info

rogerbaba committed:
set problem as warning

lucaboesch committed:
Version comment needs full stop.

lucaboesch committed:
Array brackets without space [.

lucaboesch committed:
Prepare for Moodle 4.2.

lucaboesch committed:
Prepare for Moodle 4.1.

lucaboesch committed:
All recent Moodle and PHP versions.

lucaboesch committed:
Smaller monologo icon for Moodle ≥ 4.4.

dmitriim committed:
issue #97: improve limits page

lucaboesch committed:
Prepare for Moodle 5.1

lucaboesch committed:
Tint monologo icon with dark grey as standard, resolves #110.

lucaboesch committed:
Adhere to Moodle Coding style.

lucaboesch committed:
Remove table borders in Moodle ≥ 5.0

lucaboesch committed:
Format some elements more nicely.

lucaboesch committed:
Support course overview.

lucaboesch committed:
Activity information for Moodle 5.1 activity chooser.

rogerbaba committed:
fis last commit

lucaboesch committed:
Bump version number.

rogerbaba committed:
calculation of group member count should consider setting

rogerbaba committed:
if no value is set default no validation should trigger

rogerbaba committed:
version update

rogerbaba committed:
fix hiding fullgroups also works wiht limit per group

lucaboesch committed:
limit_form.php Coding style.

rogerbaba committed:
keep it simple

rogerbaba committed:
Merge branch 'master' into activitycompletion

rogerbaba committed:
Merge pull request #116 from lucaboesch/activitycompletion

Activity completion, resolves #56

rogerbaba committed:
fix updgrade  of wrong field for completionsubmit
